build/js/live-editor.output_pjs.js

```diff
@@ -1602,10 +1614,38 @@ window.PJSOutput = Backbone.View.extend({
             // but PImage may be mutable, so that might not work.
             this.imageCache[file] = img;
         }.bind(this));
-    },
+        
+        _.each(sounds, function(file) {
+            var fileMatch = /["']([A-Za-z0-9_\/-]*?)["']/.exec(file);
 
+            if (!fileMatch) {
+                return loaded();
+            }
 
-    soundCache: {},
+            file = fileMatch[1];
+            
+            if (this.soundCache[file]) {
```

Comment:
I'd like to get your thoughts on this before proceeding:
- make the cached images look more like cached sounds, i.e. an object that looks like { image: imgElem, __id: function() { ... } }
- store all of the items in the same ```resourceCache``` where the object keys are the filenames including the extension (just incase there are some sounds with the same name as images), this will make it easy to have a single look up function
- use promises, this removes the need to count the number of resources we're going to load because we can wait for an array of promises to complete... not having to count means we can also remove the ```.match``` and just do the ```.exec```